dataprofiler/profilers/profile_builder.py

```diff
@@ -489,6 +475,22 @@ def __add__(self, other):
     def profile(self):
         return self._profile
 
+    def _get_sample_size(self, data):
+        """
+        Determines the minimum sampling size for detecting column type.
+
+        :param data: data to be profiled
+        :type data: Union[data_readers.base_data.BaseData, pandas.DataFrame]
+        :return: integer sampling size
+        :rtype: int
+        """
+        len_data = len(data)
+        if self._samples_per_update:
+            return self._samples_per_update
+        if len_data <= self._min_sample_size:
+            return int(len_data)
+        return max(int(self._sampling_ratio * len_data), self._min_sample_size)
```

Comment:
did we have a similar function for `StructuredDataProfile`? Can we use only one?

Reply:
currently we use it in both spots, however this function is slightly different as it also utilizes the samples_per_update which is unique to this class. I did not want to alter the get_sample_size in `StructuredDataProfile` as it would prolong this fix